napari/_qt/qt_main_window.py

```diff
@@ -848,7 +854,10 @@ def add_dock_widget(
         return dock_widget
 
     def _add_viewer_dock_widget(
-        self, dock_widget: QtViewerDockWidget, tabify=False, menu=None
+        self,
+        dock_widget: QtViewerDockWidget,
+        tabify: bool = False,
+        menu: Optional[Union[QModelMenu, NapariMenu]] = None,
```

Comment:
Could this be `Optional[QMenu]` instead here and in `add_dock_widget`? Both classes inherit from `QMenu` and I think the only use of the parameter is to call `add_action`, which is defined on `QMenu`.

Or is there a reason this can only be on those two types (if not `None`)?

Reply:
No reason other than I wasn't sure if putting a parent class was acceptable. Thank you, I will change.

I still think we should consider if this is functionality we want to keep with app-model but probably better discussed in an issue. I'll open one.